mqtt/tls-connect-to-broker.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: default
+title: A short (<10 words) summary of the task addressed
+slug:
+  - label: mqtt
+    url: /#mqtt
+  - secure connect
+---
+
+### Problem
+
+You want to have an encrypted connection to your Mosquitto MQTT broker.
+
+### Solution
+
+Create a valid set of certificates and keys for the broker to use.
+
+Change the configuration of the broker to start a TLS encrypted port (`mqtts`) using the above.
+
+Alter the <code class="node">MQTT Config</code> node, changing the "Server" name to start with `mqtts://`.
+
+#### Example
+
+**Valid Certificate Creation**
+
+THe easiest approach for this is to use [Let's Encrypt](). This is beyond the scopy of this article but there are plenty of examples and tutorials available on the Internet. For this to work successfully, you also need to be able to use a registered domain name on your internal network because you cannot use Let's Encrypt with IP addresses or non-public domain names.
+
+Alternatively, you can create a self-signed set of certificates. Again, this is beyond the scope of the article. However, you may need to create a trusted root certificate and provide its public cert instead of the one that Debian provides that is listed in the configuration below.
+
+**Mosquitto configuration**
+
+This assumes that you are using Let's Encrypt or other certificates signed by a root CA already trusted by the Debian operating system.
+
+Note the entries in `<...>` which need to be replaced with your own folders and files.
+
+This goes into a file of any name of the form `*.config` in the folder `/etc/mosquitto/conf.d/`. So you have to edit it with root privalages (e.g. using `sudo`).
```

Comment:
The mosquitto config files/folders are implementation specific. E.g. when using the mosquitto Docker image, by default the config file is /mosquitto/config/mosquitto.conf, which may be stored in a Docker volume or on the Docker host, depending on the container configuration.

Reply:
Added note about certificate verification.

Clarified that the example location is for direct installation on Linux.

In regard to the other, No doubt that this could be improved further but I'm not convinced that this entry should get any longer. The idea, I think is to give people a quick way to get through a standard configuration so it covers the most common scenario. Also, this entry is about connecting to the broker over mqtts, it isn't about using certificate-based authentication to the broker, I think that should be a different entry.

It is already relatively long compared to the other cookbook entries.